src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/CloudStackCloudUtils.java

```diff
@@ -49,61 +49,77 @@ public static String doRequest(@NotNull CloudStackHttpClient cloudStackHttpClien
         }
     }
 
-    // TODO (chico) - move to the CloudstackCloudUtil
+    /**
+     * Wait and process the Cloudstack asynchronous response in its asynchronous life cycle.
+     * @throws FogbowException
+     */
     public static String waitForResult(@NotNull CloudStackHttpClient client,
                                        String cloudStackUrl,
                                        String jobId,
                                        @NotNull CloudStackUser cloudStackUser)
             throws FogbowException {
-        CloudStackQueryAsyncJobResponse queryAsyncJobResult = getAsyncJobResponse(client,
-                cloudStackUrl, jobId, cloudStackUser);
-
-        if (queryAsyncJobResult.getJobStatus() == CloudStackQueryJobResult.PROCESSING) {
-            for (int i = 0; i < MAX_TRIES; i++) {
-                queryAsyncJobResult = getAsyncJobResponse(client, cloudStackUrl, jobId, cloudStackUser);
-                if (queryAsyncJobResult.getJobStatus() != CloudStackQueryJobResult.PROCESSING) {
-                    return processJobResult(queryAsyncJobResult, jobId);
-                }
-                try {
-                    Thread.sleep(ONE_SECOND_IN_MILIS);
-                } catch (InterruptedException e) {
-                    throw new FogbowException();
-                }
+
+        int countTries = 0;
+        CloudStackQueryAsyncJobResponse response = getAsyncJobResponse(
+                client, cloudStackUrl, jobId, cloudStackUser);
+        while(response.getJobStatus() == CloudStackQueryJobResult.PROCESSING) {
```

Comment:
even though == works for enum I suggest you to use .equals()

Reply:
These values are Integers. Are you suggesting to change to ENUM?

Reply:
oh, sorry. I thought they were ENUMs